### PR TITLE
[MRG] Profiling Controller

### PR DESCRIPTION
**Pull request by huangyz0918**
#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

**Comment by HuaizhengZhang** [comment]
a test example to demonstrate how to use is needed

**Comment by huangyz0918** [comment]
> a test example to demonstrate how to use is needed

Can I add in the doc?

File: modelci/controller/controller.py [hunk 1]
```diff
@@ -1,47 +1,64 @@
...
+        :param profiler: The instance of model profiler.
+        """
+        super(UtilMonitor, self).__init__()
+        self.stopped = False
+        self.delay = delay
+        self.memory_level = memory_level
+        self.util_level = util_level
+        self.profiler = profiler
+
+        if self.exporter is None:
```
Comment (HuaizhengZhang) [inline modelci/controller/controller.py:29]: where is it defined?

File: modelci/controller/controller.py [hunk 2]
```diff
@@ -1,47 +1,64 @@
...
+            raise TypeError(
+                'Failed to get GPU relative information from node exporter, please make sure the service has started.')
+
+    def run(self):
+        while not self.stopped:
+            exporter = GPUNodeExporter()
```
Comment (HuaizhengZhang) [inline modelci/controller/controller.py]: repeatedly instantiate the exporter?

File: modelci/controller/controller.py [hunk 3]
```diff
@@ -1,47 +1,64 @@
...
+            available_device = exporter.get_idle_gpu(util_level=self.util_level,
+                                                     memory_level=self.memory_level)
+            if available_device is not None:
+                profiler.auto_diagnose(available_devices=available_device,
+                                       batch_list=[8])  # default profiling batch size 8
```
Comment (HuaizhengZhang) [inline modelci/controller/controller.py]: only profile batch_size = 8?

auto_diagnose should profile models under different batch sizes

**Comment by huangyz0918** [comment]
Actually, it is very easy to understand how to use, just calling `auto_model_profiling`, nothing else. Maybe we can add a completed example when all the controller is ready.

File: modelci/controller/controller.py [hunk 4]
```diff
@@ -1,47 +1,64 @@
...
+            time.sleep(self.delay)
+
+    def stop(self):
+        self.stopped = True
+
+
```
Comment (HuaizhengZhang) [inline modelci/controller/controller.py:46]: https://github.com/anderskm/gputil

better use this toolkit to get the device type here

later we can move it to the exporter class

**Comment by HuaizhengZhang** [comment]
@YuanmingLeee please check. If no question for you, Yizheng can merge this.

**Pull request merged**